src/core/positioning/egenioussreceiver.cpp

```diff
@@ -0,0 +1,124 @@
+#include "egenioussreceiver.h"
+
+#include <QHostAddress>
+#include <QJsonDocument>
+#include <QJsonValue>
+
+EgenioussReceiver::EgenioussReceiver( QObject *parent )
+  : AbstractGnssReceiver( parent ), mTcpSocket( new QTcpSocket() )
+{
+  connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
+  connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::handleError );
+  connect( mTcpSocket, &QTcpSocket::connected, this, &EgenioussReceiver::connected );
+  connect( mTcpSocket, &QTcpSocket::disconnected, this, &EgenioussReceiver::disconnected );
+}
+
+void EgenioussReceiver::handleConnectDevice()
+{
+  mTcpSocket->connectToHost( mAddress, mPort, QTcpSocket::ReadWrite );
+}
+
+void EgenioussReceiver::connected()
+{
+  mSocketState = QAbstractSocket::ConnectedState;
+  emit socketStateChanged( mSocketState );
+  setValid( true );
+}
+
+void EgenioussReceiver::handleDisconnectDevice()
+{
+  mTcpSocket->disconnectFromHost();
+}
+
+void EgenioussReceiver::disconnected()
+{
+  if ( mTcpSocket->state() == QAbstractSocket::ConnectedState )
+  {
+    mSocketState = QAbstractSocket::UnconnectedState;
+    emit socketStateChanged( mSocketState );
+  }
+}
+
+QList<QPair<QString, QVariant>> EgenioussReceiver::details()
+{
+  QList<QPair<QString, QVariant>> detailsList;
+
+  if ( mPayload.isEmpty() )
+  {
+    return detailsList;
+  }
+
+  detailsList.append( qMakePair( "q", mPayload.value( "q" ).toDouble() ) );
+
+  return detailsList;
+}
+
+void EgenioussReceiver::onReadyRead()
+{
+  QByteArray mReceivedData = mTcpSocket->readAll();
+  if ( mReceivedData.size() < 9 )
```

Comment:
```suggestion
  if ( mReceivedData.size() < EgenioussReceiver::MINIMUM_DATA )
```
I would define consts for such kind of things

Reply:
Good idea